build-system/pr-check.js

```diff
@@ -286,16 +286,17 @@ function runAllCommands() {
     command.runVisualDiffTests();
     command.runJsonAndLintChecks();
     command.runDepAndTypeChecks();
-    command.runUnitTests();
-    // command.testDocumentLinks() is skipped during push builds.
     command.buildValidatorWebUI();
     command.buildValidator();
+    command.runUnitTests(); // Requires validator to be built.
+    // command.testDocumentLinks() is skipped during push builds.
   }
   if (process.env.BUILD_SHARD == "integration_tests") {
     command.cleanBuild();
     command.buildRuntimeMinified();
     command.runPresubmitTests();  // Needs runtime to be built and served.
-    command.runIntegrationTests();
+    command.buildValidator();
```

Comment:
By always doing this in both shards (instead of just in the unit_tests shard and only for validator changes), we're adding an extra 2m 30s to the Travis build cycle. I'm guessing there's no way to avoid this, right?